lib/CGI/Application.pm

```diff
@@ -1078,12 +1078,36 @@ support to it.
 The simplest way to create and return a PSGI-compatible coderef. Pass in
 arguments to a hashref just as would to new. This returns a PSGI-compatible
 coderef, using L<CGI:::PSGI> as the query object. To use a different query
-object, construct your own object using C<< run_as_psgi() >>, as shown below.
+object, you have to construct your own object using C<< run_as_psgi() >>, 
+as shown below, because C<< psgi_app >> creates a new CGI::PSGI object
+unconditionally on every request and therefore overrides a own query object.
```

Comment:
Initial version looked better IMHO. The only thing that should be added/highlighted is that `QUERY` parameter is not supported and will be overwritten.

I would even add a warning / exception in the code if we detect that it was passed.